curriculum/challenges/english/25-front-end-development/lecture-working-with-state-and-responding-to-events-in-react/67d1eaf77f510e15dfed7c14.md

```diff
@@ -2,13 +2,130 @@
+
+Let's look into what state is and how the `useState` hook lets you work with it.
+
+State represents the dynamic data in your React component, like the value from a user input, data fetched from an API, or an item in a to-do list.
+
+Whenever the state changes, React re-renders the component without reloading the page to reflect those changes in the User Interface. This reactivity makes your app interactive.
+
+The `useState` hook is a function that lets you declare state variables in functional components. 
+
+Before hooks, you could only use state in class components. But with the introduction of hooks since React 16.8, you can use state in functional components by using the `useState` hook.
+
+To use the `useState` hook, you need to import it from react:
+
+```js
+import { useState } from "react";
+```
+
+You can also import React itself and get access to the `useState` hook as a property:
+
+```js
+import React from "react";
+```
+
+Here's how you can declare a state variable when you import `useState`:
+
+```js
+const [stateVariable, setStateFunction] = useState(initialValue);
+```
+
+And here's how you can declare a state variable when you import React:
+
+```js
+const [stateVariable, setStateFunction] = React.useState(initialValue);
+```
+
+In the state variable you have the following:
+
+- `stateVariable` holds the current state value
+- `setStateFunction` (the setter function) updates the state variable
+- `initialValue` sets the initial state
+
+Note that the state in a React component is private, and is isolated to each component instance. This means that, if you render the same component twice, the state component of one does not affect the other. This also means that, if you'd like to share state between components, then you'd need to lift the state up to a common parent and pass it down as props.
+
+Another thing is that hooks must be called at the top level of a component, just before the return keyword, to keep the state and effects consistent across renders. This means you can't use state inside loops, conditions, or nested functions.
+
+Here's an example of managing state with the `useState` hook in a `Counter` component:
+
+```jsx
+// Importing the useState hook
+import { useState } from "react";
+
+function Counter() {
+  const initialValue = 0;
+
+  // The state variable and setter function
+  const [count, setCount] = useState(initialValue);
+
+  return (
+    <div>
+      {/* Display current state value */}
+      <h2>{count}</h2>
+
+      <button onClick={() => setCount(count - 1)}>Decrement</button>
+      <button onClick={() => setCount(count + 1)}>Increment</button>
+    </div>
+  );
+}
+
+export default Counter;
+```
+
+In the code above, we have the `useState` hook imported from React, and the `Counter` state and its setter initialized with an initialValue of `0`. The return statement contains the count and two buttons to decrement and increment the count by `1`.
```

Comment:
```suggestion
In the code above, we have the `useState` hook imported from React, and the `Counter` state and its setter (`setCount`) initialized with an initial value of `0`. The `return` statement contains the count and two buttons to decrement and increment the count by `1`.
```
Honestly, I think this sentence is not clear. Is it correct to say that the setter (`setCount`) is initialized to an initial value? And why the state variable is not mentioned explicitly?

Reply:
> Is it correct to say that the setter (setCount) is initialized to an initial value?

I think it is trying to mention that you have both state and a setter function. At least that is how I read it.
You could update it to say the following:

```md
In the code above, we have the `useState` hook imported from React. In the `Counter` component, the `count` represents the current state while `setCount`  is the set function responsible for updating state.  The current state value is `0`. The `return` statement contains the count and two buttons to decrement and increment the count by `1`.
```

Reply:
```suggestion
In the code above, we have the `useState` hook imported from React. In the `Counter` component, the `count` represents the current state while `setCount` is the set function responsible for updating state. The current state value is `0`. The `return` statement contains the count and two buttons to decrement and increment the count by `1`.
```
I like that :+1: